File: benchmark/php/vector_dot.php
<?php
$a = []; $b = [];
for ($i = 0; $i < 500000; $i++) { $a[] = $i + 1; $b[] = $i + 2; }
$start = hrtime(true);
$sum = 0;
for ($i = 0; $i < 500000; $i++) { $sum += $a[$i] * $b[$i]; }
$end = hrtime(true);
$ms = ($end - $start) / 1e6;
echo "Dot: " . $sum . "
";
echo "Time: " . round($ms) . "ms
";

File: benchmark/python/vector_dot.py
import time

a = []
b = []
for i in range(500000):
    a.append(i + 1)
    b.append(i + 2)
sum = 0
start = time.time() * 1000
for i in range(500000):
    sum = sum + a[i] * b[i]
end = time.time() * 1000
print(f"Dot: {sum}")
print(f"Time: {end - start:.0f}ms")
